movieland_service_impl/src/main/java/com/solovey/movieland/service/impl/ReportingServiceImpl.java

```diff
@@ -0,0 +1,233 @@
+            }
+
+            for (Report report : reportsMetadataCache.values()) {
+                if (report.getUserId() == userId) {
+                    userReports.add(report);
+                }
+            }
+
+        } finally {
+            lock.readLock().unlock();
+        }
+        return userReports;
+    }
+
+    @Override
+    public Report getReportMetadata(String reportId, int userId) {
+        Report report = reportsMetadataCache.get(reportId);
+        if (report != null && report.getReportState() == ReportState.READY && report.getUserId() == userId) {
+            return report;
+        }
+        throw new ReportNotFoundException();
+    }
+
+    @Override
+    public void deleteReport(String reportId, int userId) {
+        lock.writeLock().lock();
+        try {
+            Iterator<Report> iterator = reportRequests.iterator();
+            while (iterator.hasNext()) {
+                Report report = iterator.next();
+                if (report.getReportId().equals(reportId)) {
+                    if (report.getUserId() == userId) {
+                        iterator.remove();
+                        log.info("Report {} has been removed from queue", report.getReportId());
+                        return;
+                    } else {
+                        throw new SecurityException("You cannot remove this report");
+                    }
+                }
+            }
+        } finally {
+            lock.writeLock().unlock();
+        }
+
+        Report report = reportsMetadataCache.get(reportId);
+        if (report != null) {
+            if (report.getUserId() != userId) {
+                throw new SecurityException("You cannot remove this report");
+            }
+            reportsMetadataCache.remove(reportId);
+
+            if (report.getReportState() != ReportState.IN_PROGRESS) {
+                deleteReportFile(report);
+                reportingDao.removeReportMetadata(reportId);
+                log.info("Report {} has been removed", report.getReportId());
+            } else {
+                fixedThreadPoolExecutor.submit(() -> {
+                    try {
+                        report.getCountDownLatch().await();
+                        deleteReportFile(report);
+                    } catch (InterruptedException e) {
+                        log.warn("thread that deletes report is interrupted");
+                    }
+
+                    reportingDao.removeReportMetadata(reportId);
+                    log.info("Report {} has been removed", report.getReportId());
+                });
+
+            }
+        }
+
+
+    }
+
+    private void deleteReportFile(Report report) {
+        String fileName = report.getPath();
+        File file = new File(fileName);
+        file.delete();
+    }
+
+    @Scheduled(fixedDelayString = "${reporting.sheduler.interval}",
+            initialDelayString = "${reporting.sheduler.interval}")
+    private void processRequests() {
```

Comment:
I didn't find Scheduled task to execute claeanup of Reporting cache on annual basis
>> Requests for reports are checked by scheduled job every 5 minutes.
>> Generated reports should be cleaned by scheduled job every Monday of July 15:15:15 ;)